### PR TITLE
TypeTransformer for Keras

### PR DESCRIPTION
**Pull request by ryankarlos**
Signed-off-by: Ryan Nazareth <ryankarlos@gmail.com>

# TL;DR
Adds a typetransformer to support `keras.Model` and `keras.Sequential` as native types

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description

* `__init__.py` file handles case where the user doesn't have keras installed.
* TypeTransformer implemented for the following classes: Sequential: https://keras.io/api/models/sequential/
* added tests in  https://github.com/flyteorg/flytekit/tree/master/tests/flytekit/unit/extras/keras

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2759

## Follow-up issue
N/A

### CODE REVIEW

File: dev-requirements.in [hunk 1]
```diff
@@ -13,3 +13,4 @@ google-cloud-bigquery
 google-cloud-bigquery-storage
 IPython
 torch
+tensorflow<=2.8.1
```
Comment (ryankarlos) [inline dev-requirements.in:16]: see my comment in https://github.com/flyteorg/flytekit/pull/1240#discussion_r1000932914 for reasons for pinning tf versions.

Comment (ryankarlos) [reply dev-requirements.in]: Ignore this - ive pinned `grpcio-status<1.49.0` instead based on suggestion from @pingsutw  in another PR

**Review by pingsutw — COMMENTED** [review]
There is an error in `test_to_python_value_and_literal`
```python
    def test_to_python_value_and_literal(transformer, python_type, format, python_val):
        ctx = context_manager.FlyteContext.current_context()
        tf = transformer
        lt = tf.get_literal_type(python_type)
        lv = tf.to_literal(ctx, python_val, type(python_val), lt)  # type: ignore
        assert lv.scalar.blob.metadata == BlobMetadata(
            type=BlobType(
                format=format,
                dimensionality=BlobType.BlobDimensionality.SINGLE,
            )
        )
        assert lv.scalar.blob.uri is not None
    
        output = tf.to_python_value(ctx, lv, python_type)
        if isinstance(python_val, keras.Sequential):
            for p1, p2 in zip(output.weights, python_val.weights):
                np.testing.assert_array_equal(p1.numpy(), p2.numpy())
            assert True
        else:
>           assert isinstance(output, dict)
E           assert False
E            +  where False = isinstance(<keras.engine.functional.Functional object at 0x000001AE37320108>, dict)
```

**Comment by ryankarlos** [comment]
> There is an error in `test_to_python_value_and_literal`
> 
> ```python
>     def test_to_python_value_and_literal(transformer, python_type, format, python_val):
>         ctx = context_manager.FlyteContext.current_context()
>         tf = transformer
>         lt = tf.get_literal_type(python_type)
>         lv = tf.to_literal(ctx, python_val, type(python_val), lt)  # type: ignore
>         assert lv.scalar.blob.metadata == BlobMetadata(
>             type=BlobType(
>                 format=format,
>                 dimensionality=BlobType.BlobDimensionality.SINGLE,
>             )
>         )
>         assert lv.scalar.blob.uri is not None
>     
>         output = tf.to_python_value(ctx, lv, python_type)
>         if isinstance(python_val, keras.Sequential):
>             for p1, p2 in zip(output.weights, python_val.weights):
>                 np.testing.assert_array_equal(p1.numpy(), p2.numpy())
>             assert True
>         else:
> >           assert isinstance(output, dict)
> E           assert False
> E            +  where False = isinstance(<keras.engine.functional.Functional object at 0x000001AE37320108>, dict)
> ```

Ahh yes, forgot to account for keras.Model in the check ....... just pushed fix now
